Application/src/view/visuals/component/Message.java

```diff
@@ -231,11 +237,26 @@ public void setStatic(boolean staticIndicator)  {
     public void renderDefault(GraphicsContext gc) {
 
 
+        if(trails.size() != 0) {
+            Trail tempTrail;
+            //Draws all trails except the first and last one
+            for (int i = 1; i < trails.size() - 1; i++) {
+                tempTrail = trails.get(i);
+                gc.drawImage(cloud, tempTrail.getXcoordinate(), (tempTrail.getYcoordinate() + 18), tempTrail.getWidth(), tempTrail.getHeight());
+            }
 
-        //Draw trail for the message, for each instance in the arraylist
-        for (Trail t: trails)
-            gc.drawImage(trail, t.getXcoordinate(), (t.getYcoordinate() + 18), t.getWidth(), t.getHeight());
-
+            int trailSize = trails.size() - 1;
+            if(trailSize != 0) {
+                Trail last = trails.get(trailSize);
+                //Puts an arrow on the last location of the trail array
+                //The if statement changes the direction of the arrow if necessary
+                if (directionSwitched) {
```

Comment:
The complexity is unneccessary with this. Would it be possible to move the functionality of changing of arrow-imgs to outside of this, in order to avoid repetative code? 
lines 253-259 are the same. See how Pontus did this with the "animation" of the castle flag

Reply:
@IsabelleTornqvist Animation? i am using a boolean to decide which arrow to use, so i am still gonna 2 different draw image lines
i could make a new method that rotates the arrow but that would unnecessary since there is only 2 states and it would increase coupling and complexity between methods

Reply:
Roguhly something like this:
if(directionSwitched){
tempImg = flippedArrow;
}else{
tempImg = arrow;
}
 gc.drawImage(arrow, last.getXcoordinate(), (last.getYcoordinate() + 18), last.getWidth(), last.getHeight());

Reply:
So that we don't get the repetition of:
 gc.drawImage(arrow, last.getXcoordinate(), (last.getYcoordinate() + 18), last.getWidth(), last.getHeight());

Reply:
@IsabelleTornqvist i tried it, didn't work because images aren't stored like that, 
feel free to add a commit yourself to the branch or dismiss this unnecessary change

Reply:
gc.drawImage(directionSwitched?flippedArrow:arrow,x,y,w,h);
also works.

Reply:
@pontuslaestadius yes that actually works!